### PR TITLE
Filter out Prometheus scrape_ metrics if EnhancedContainer Insights is enabled

### PR DESCRIPTION
**Pull request by mitali-salvi**
**Description:**
Filter out scrape_ metric namely - "up", "scrape_duration_seconds", "scrape_samples_post_metric_relabeling", "scrape_samples_scraped", "scrape_series_added" taken from https://prometheus.io/docs/concepts/jobs_instances when enhancedContainerInsights is enabled in the EMF exporter

**Testing:** 
Screenshot 2023-10-10 at 12 21 13

Screenshot 2023-10-10 at 14 24 06

### CODE REVIEW

File: exporter/awsemfexporter/emf_exporter.go [hunk 1]
```diff
@@ -107,6 +107,12 @@ func (emf *emfExporter) pushMetricsData(_ context.Context, md pmetric.Metrics) e
 	for _, groupedMetric := range groupedMetrics {
 		cWMetric := translateGroupedMetricToCWMetric(groupedMetric, emf.config)
 		putLogEvent, err := translateCWMetricToEMF(cWMetric, emf.config)
+		if emf.config.EnhancedContainerInsights && (putLogEvent != nil &&
+			putLogEvent.InputLogEvent != nil &&
+			putLogEvent.InputLogEvent.Message != nil) && *putLogEvent.InputLogEvent.Message == "" {
+			emf.config.logger.Info("Dropping Prometheus job instance metrics log for EnhancedContainerInsights")
```
Comment (chadpatel) [inline exporter/awsemfexporter/emf_exporter.go:113]: is this info message too specific?  This does not seem to do anything prometheus specific, it looks like it just drops inputs where the message is ""

--

Should this be debug? IIRC the other `dropping` log line is a debug log.

File: exporter/awsemfexporter/metric_translator.go [hunk 2]
```diff
@@ -358,6 +367,24 @@ func translateCWMetricToEMF(cWMetric *cWMetrics, config *Config) (*cwlogs.Event,
...
+	for key := range fieldMap {
+		if _, ok := prometheusInstanceScrapeMetrics[key]; ok {
+			config.logger.Debug(
+				"Detected prometheus job instance metrics",
+				zap.Any("key", key),
+				zap.Any("value", fieldMap[key]),
+			)
+			// Return empty logEvent for prometheusInstanceScrapeMetrics if EnhancedContainerInsights is enabled
+			if config.EnhancedContainerInsights {
+				logEvent := cwlogs.NewEvent(
```
Comment (sky333999) [inline exporter/awsemfexporter/metric_translator.go:379]: Why return an empty log event? We could have other fields that customers might need in the same event.

Cant we just drop the specific fields?

File: exporter/awsemfexporter/metric_translator.go [hunk 3]
```diff
@@ -31,6 +31,15 @@ const (
 	fieldPrometheusMetricType = "prom_metric_type"
 )
 
+// Prometheus instance availability monitoring metrics - https://prometheus.io/docs/concepts/jobs_instances
+var prometheusInstanceScrapeMetrics = map[string]struct{}{
```
Comment (sky333999) [inline exporter/awsemfexporter/metric_translator.go:35]: Id just make this a configurable feature if the exporter.
Allow specifying `dropFields`.

File: exporter/awsemfexporter/emf_exporter.go (hunk 1 above)
Comment (mitali-salvi) [reply exporter/awsemfexporter/emf_exporter.go:113]: Updated the log line to debug

**Review by chadpatel — APPROVED** [review]

File: exporter/awsemfexporter/emf_exporter.go [hunk 4]
```diff
@@ -107,6 +107,12 @@ func (emf *emfExporter) pushMetricsData(_ context.Context, md pmetric.Metrics) e
 	for _, groupedMetric := range groupedMetrics {
 		cWMetric := translateGroupedMetricToCWMetric(groupedMetric, emf.config)
 		putLogEvent, err := translateCWMetricToEMF(cWMetric, emf.config)
+		if emf.config.EnhancedContainerInsights && (putLogEvent != nil &&
```
Comment (sky333999) [inline exporter/awsemfexporter/emf_exporter.go:110]: As discussed, might be better to just return nil below and check for putLogEvent == nil here.

Separately, we should move this to below the err check as this shouldnt make any errors.

**Review by chadpatel — APPROVED** [review]

**Review by sky333999 — APPROVED** [review]

**Pull request merged**